packages/autocertifier/test/end-to-end/clientServer.test.ts

```diff
@@ -0,0 +1,175 @@
+import { 
+    ConnectionManager,
+    ListeningRpcCommunicator,
+    NodeType,
+    PeerDescriptor,
+    PeerID,
+    Simulator,
+    SimulatorTransport,
+    DefaultConnectorFacade
+} from '@streamr/dht'
+import { AutoCertifierClient, createSelfSignedCertificate } from '@streamr/autocertifier-client'
+import os from 'os'
+import fs from 'fs'
+import { RestServer } from '../../src/RestServer'
+import { CertifiedSubdomain } from '@streamr/autocertifier-client'
+import { Session } from '@streamr/autocertifier-client'
+import { v4 } from 'uuid'
+import { Logger } from '@streamr/utils'
+import { StreamrChallenger } from '../../src/StreamrChallenger'
+import { SessionIdRequest, SessionIdResponse } from '../../src/proto/packages/autocertifier/protos/AutoCertifier'
+
+const logger = new Logger(module)
+
+let server: RestServer
+const dir = os.tmpdir()
+let certifiedSubdomain: CertifiedSubdomain
+
+const createTestSubdomain = (validityMillis?: number) => {
+    if (validityMillis) {
+        const fakeCerts = createSelfSignedCertificate('localhost', 1200)
+        certifiedSubdomain = { 
+            fqdn: 'localhost',
+            subdomain: 'fwefwafeaw',
+            token: 'token',
+            certificate: {
+                cert: fakeCerts.serverCert,
+                key: fakeCerts.serverKey
+            }
+        }
+    } else {
+        const fakeCerts = createSelfSignedCertificate('localhost', 0, validityMillis)
+        certifiedSubdomain = { 
+            fqdn: 'localhost',
+            subdomain: 'fwefwafeaw',
+            token: 'token',
+            certificate: {
+                cert: fakeCerts.serverCert,
+                key: fakeCerts.serverKey
+            }
+        }
+    }
+}
+
+describe('clientServer', () => {
+
+    const restServerPort = '3000'
+
+    const mockPeerDescriptor1: PeerDescriptor = {
+        kademliaId: PeerID.fromString('tester1').value,
+        type: NodeType.NODEJS
+    }
+   
+    const simulator = new Simulator()
+    const mockTransport = new SimulatorTransport(mockPeerDescriptor1, simulator)
+
+    let clientConnectionManager: ConnectionManager
+    let clientRpcCommunicator: ListeningRpcCommunicator | undefined
+
+    const subdomainPath = os.tmpdir() + 'subdomain.json'
+
+    let client: AutoCertifierClient
+
+    beforeAll(async () => {
+
+        if (fs.existsSync(subdomainPath)) {
+            fs.unlinkSync(subdomainPath)
+        }
+        createTestSubdomain()
+        server = new RestServer('localhost', '127.0.0.1', restServerPort, dir + '/restServerCaCert.pem', dir + '/restServerCaKey.pem',
```

Comment:
Do these files exists somewhere, or how the file is generated with that name?

Reply:
The files are created by the `RestServer` if they do not exist.